src/main/java/week5/AlphabetFinderTest.java

```diff
@@ -0,0 +1,102 @@
+package week5;
+
+
+import static org.hamcrest.core.Is.is;
+import static org.junit.Assert.assertThat;
+
+import java.util.Arrays;
+import java.util.Collections;
+import org.junit.Test;
+
+public class AlphabetFinderTest {
+
+  @Test
+  public void testEmpty() {
+    assertThat(AlphabetFinder.findAlphabet(null), is(Collections.EMPTY_LIST));
+    assertThat(AlphabetFinder.findAlphabet(new String[]{}), is(Collections.EMPTY_LIST));
+  }
+
+  @Test
+  public void testExample() {
+    //test example that was given
+    assertThat(AlphabetFinder.findAlphabet(new String[]{"ART", "RAT", "CAT", "CAR"}),
+        is(Arrays.asList('A', 'T', 'R', 'C')));
+  }
+
+  @Test
+  public void testContradiction() {
+    //does the program crash on contradiction
+    assertThat(AlphabetFinder.findAlphabet(new String[]{"ART", "RAT", "ARC", "CAT", "CAR"}),
+        is(Arrays.asList('A', 'T', 'R', 'C')));
+  }
+
+  @Test
+  public void testMergeCorrection() {
+    //partial order refines alphabet - there is a letter that should be moved forward
+    assertThat(AlphabetFinder.findAlphabet(new String[]{"ART", "RAT", "CTT", "CTA", "CRA"}),
+        is(Arrays.asList('T', 'A', 'R', 'C')));
+  }
+
+  @Test
+  public void testMergeCorrectionB() {
+    //partial order refines alphabet - there is a letter that should be moved backward
+    assertThat(AlphabetFinder.findAlphabet(new String[]{"AC", "DR", "D", "TAR", "TAC", "TAD"}),
+        is(Arrays.asList('A', 'R', 'C', 'D', 'T')));
+  }
+
+  @Test
+  public void testAmbiguous() {
+    //partial order is more ambigious than alphabet
+    assertThat(AlphabetFinder.findAlphabet(new String[]{"ART", "RAT", "CAT", "CAA", "DT", "DC"}),
+        is(Arrays.asList('T', 'A', 'R', 'C', 'D')));
```

Comment:
No what I mean is that the last partial order [T,C] from DT, DC does not add more information as the position of T is already established from CAT, CAA